No instance for ‘MonadTrans TransformT’

### Description
Hi,
when updating [`retrie`](https://hackage.haskell.org/package/retrie) to support mtl 2.3 (https://github.com/facebookincubator/retrie/commit/454443c03918a1a211a1df2203916275f9314340), I got the following errors:
```
[ 1 of 14] Compiling Annotated        ( tests/Annotated.hs, dist/build/test/test-tmp/Annotated.dyn_o )

tests/Annotated.hs:121:5: error: [GHC-39999]
    • No instance for ‘MonadTrans TransformT’
        arising from a use of ‘lift’
    • In the first argument of ‘($)’, namely ‘lift’
      In a stmt of a 'do' block: lift $ liftIO $ assertGoodSrcSpan ss
      In the expression:
        do ss <- transformWithSeedIncreaseCheck uniqueSrcSpanT
           lift $ liftIO $ assertGoodSrcSpan ss
    |
121 |     lift $ liftIO $ assertGoodSrcSpan ss
    |     ^^^^

tests/Annotated.hs:134:3: error: [GHC-39999]
    • No instance for ‘MonadTrans TransformT’
        arising from a use of ‘lift’
    • In the first argument of ‘($)’, namely ‘lift’
      In a stmt of a 'do' block:
        lift
          $ liftIO $ assertBool "transform increases seed" (seed' > seed)
      In the expression:
        do seed <- get
           x <- m
           seed' <- get
           lift
             $ liftIO $ assertBool "transform increases seed" (seed' > seed)
           ....
    |
134 |   lift $ liftIO $ assertBool "transform increases seed" (seed' > seed)
    |   ^^^^
```
### Analysis
This seems related to the removal of `MonadTrans` from `TransformT` as done in https://github.com/alanz/ghc-exactprint/commit/db5e8ab3817c9ee34e37359d5839e9526e05e448#diff-1b94ad90666f59404d706ba788358ff34a88b8994191da42e6c878ac83995cd9
The related PR also explicitly mentions this change and reasons that there was a warning that it was unnecessary: https://github.com/alanz/ghc-exactprint/pull/124#discussion_r1140998162

### Possible solutions
Reverting the change leads to `retrie` building fine (with above patch to support mtl 2.3). There is also no breakage in other packages afaict and the tests pass either way. I'll make a PR for that in short order.
I also didn't encounter any warnings during building.

### System information
OS: Arch Linux
Linux: 6.6.7-arch1-1
GHC: 9.6.3
ghc-exactprint: 1.7.0.1
retrie: 1.2.2 (with commit for `mtl 2.3` support)

--- a/src/Language/Haskell/GHC/ExactPrint/Transform.hs
+++ b/src/Language/Haskell/GHC/ExactPrint/Transform.hs
@@ -119,6 +119,7 @@ newtype TransformT m a = TransformT { unTransformT :: RWST () [String] Int m a }
                          ,MonadReader ()
                          ,MonadWriter [String]
                          ,MonadState Int
+                         ,MonadTrans
                          )
 
 instance Fail.MonadFail m => Fail.MonadFail (TransformT m) where

